### ISSUE TITLE
Cannot index notebooks without adding this line

### ISSUE THREAD

**Issue by shuwu-db** [elsevierlabs-os/NotebookDiscovery#2]
<img width="1210" alt="Screen Shot 2021-12-13 at 3 08 12 PM" src="https://user-images.githubusercontent.com/84782979/145767724-f390aa83-cc1c-41cb-b519-796cf424e769.png">
 
Hi Elsevier team, thank you for creating this awesome notebook to compliment the search function inside Databricks. I have introduced this project to several of my customers and they all found the indexed parquet tables will have no results. I took a look at the code and found adding this line (as screenshot) will make it work as normal. It could be because the returned results from API are different now? The first entry of the zip file seems to be the manifest file, which is not useful for searching, causing the later table parsing to have no results in the schema. Can you take a look? Thanks!

**Comment by dmcbeath** [comment]
Thanks for your interest and happy to take a look.  Could you provide some more details on the configuration you are running (DBR, Cloud provider, etc.). We have not noticed the behavior you describe.  Typically, there are no results to index when there is a permissions problem (the person running the job doesn't have the necessary permissions to read the folders/files).  But, this doesn't appear to be the case.

**Comment by shuwu-db** [comment]
Thanks for the reply. I am running on DBR 9.1 LTS, and I am one of the admins on the Azure Databricks workspace. My customer is also running on DBR 9.1 LTS on an Azure Databricks workspace. I have checked I can see everyone's notebook (which is expected for an admin). I have walkthrough it step-by-step. I can see all of the notebooks are found from the walkPath function. However, in the exportNotebook function, it only retrieves the first element of the array, which is the manifest file, while the second element is the actual file code where we can find cLang, cText etc.
<img width="1306" alt="Screen Shot 2021-12-14 at 8 57 41 AM" src="https://user-images.githubusercontent.com/84782979/145913387-91d1a9a1-eea7-4b32-90b0-86d37616bec8.png">

If I print out what is the output for the notebookContent without adding the line, it looks like this which will later make the dataframe parsing fail. This also happens to my other customer who dropped me this question of why they are seeing no index files...It looks like the API call returns an array now that has two elements. The first one looks like the printout below, and the other one is the correct one?
<img width="948" alt="Screen Shot 2021-12-14 at 9 03 07 AM" src="https://user-images.githubusercontent.com/84782979/145913622-85981dcd-e517-4eba-bbbf-6cf721d1476a.png">

**Comment by dmcbeath** [comment]
Thanks for the detailed information.  Let me look a bit closer to see what might be happening (and why I'm not seeing this behavior).  I'll post an update in the next couple of days.

**Comment by dmcbeath** [comment]
ok.  I was able to confirm that I'm seeing this on my end as well (across many DBRs (7.* to 10.*).  This is really quite strange because this has worked in the past (worked back on Oct 27 when I ran it with either DBR 8.* or 9.*) and something appears to have changed with the DBC that is being returned for the REST API export call.  Either a manifest file is now being returned or the order of them has switched and I was just getting lucky by taking the first entry.  In either case, I will plan to get a change pushed tomorrow.  Thanks again for bringing this to my attention and for the detailed explanation of the problem.

**Comment by shuwu-db** [comment]
Thanks you Darin. Yeah I think Databricks might have made some changes to the DBC format or something else changed in this month. It was working as always until the customer asked me that their index file is empty. Probably we can make the extraction more robust. Thanks again!! (amazing companion tools for Databricks users)

**Comment by dmcbeath** [comment]
Ok, I have just pushed NotebookDiscovery-1.0.2.dbc to address this problem.

There appears to have been a change in the DBC Export format from a REST API. Either a manifest was added or the order of the files (manifest and notebook contents) was changed. In any case, changes have been made to make the solution more robust.

We now set the notebook to the zipentry string if the entry is for a notebook (specifically a V1 notebook). For example:

if (zipEntryJson.has("version") && zipEntryJson.getString("version") == "NotebookV1")

This was tested on DBR 8.*, 9.*, and 10.*.  Let me know if you still experience issues.
